ros2_batch_job/__main__.py

```diff
@@ -276,6 +291,51 @@ def get_args(sysargv=None):
     return args
 
 
+def include_python_coverage_in_report(args, package=None):
+    # no arguments means report all packages in buildspace
+    root_directory_to_report = args.buildspace
+    if package:
+        root_directory_to_report = os.path.join(root_directory_to_report, package)
+
+    coverage_xml_files = glob.glob(
+        os.path.join(root_directory_to_report, '**', 'coverage.xml'),
+        recursive=True)
+    for coverage_xml_path in coverage_xml_files:
+        # python coverage detected: copy the coverage.xml file to buildspace to be reported
+        dest_file_to_report = os.path.join(args.buildspace, coverage_xml_path.replace(os.path.sep, '.'))
+        copyfile(coverage_xml_path, dest_file_to_report)
+
+
+def filter_unit_coverage(args, coverage_info_file):
+    build_paths_collection = []
+    for package_name in args.coverage_filter_packages:
+        # check if it is a python package generating its own coverage.xml
+        include_python_coverage_in_report(args, package_name)
+        build_paths_collection.append('*%s/*' % (str(os.path.join(args.buildspace, package_name))))
+
+    # collect paths to run lcov in order to process C/C++ coverage information
+    cmd = [
+        args.colcon_script,
+        'list',
+        '--paths-only',
+        '--base-paths', args.sourcespace,
+        '--packages-select'] + args.coverage_filter_packages
+    print(cmd)
+    src_paths_collection = subprocess.check_output(cmd).decode('ascii').strip().splitlines()
+    if len(src_paths_collection) != len(args.coverage_filter_packages):
+            print('One of the packages to report coverage rate for was not found')
```

Comment:
Error messages should go to `stderr`. It might also be helpful to include which package names were missing in the message.